src/Lykke.Service.OperationsHistory.Job/RabbitSubscribers/MongoDbSubscriber.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Linq;
+using System.Threading.Tasks;
+using Autofac;
+using Common;
+using Common.Log;
+using Lykke.RabbitMqBroker;
+using Lykke.RabbitMqBroker.Subscriber;
+using Lykke.Service.ClientAccount.Client;
+using Lykke.Service.ClientAccount.Client.AutorestClient;
+using Lykke.Service.ClientAccount.Client.AutorestClient.Models;
+using Lykke.Service.OperationsHistory.Core;
+using Lykke.Service.OperationsHistory.Core.Services;
+using Lykke.Service.OperationsHistory.Core.Settings.Job;
+using Lykke.Service.OperationsHistory.Mongo;
+using Lykke.Service.OperationsRepository.Contract.History;
+using Microsoft.Extensions.Caching.Distributed;
+using Lykke.Service.OperationsRepository.Contract;
+
+namespace Lykke.Service.OperationsHistory.Job.RabbitSubscribers
+{
+    public class MongoDbSubscriber : IStartable, IStopable
+    {
+        private readonly ILog _log;
+        private RabbitMqSubscriber<OperationsHistoryMessage> _subscriber;
+        private readonly RabbitMqSettings _rabbitSettings;
+        private readonly IOperationsHistoryRepository _operationsHistoryRepository;
+        private readonly IDistributedCache _distributedCache;
```

Comment:
I would use in-memory implementation of IDistributedCache here for better performance. Or "OnDemandDataCache".